core/src/main/java/com/graphhopper/routing/util/WheelchairFlagEncoder.java

```diff
@@ -0,0 +1,312 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.util;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.profiles.EncodedValue;
+import com.graphhopper.routing.profiles.UnsignedDecimalEncodedValue;
+import com.graphhopper.storage.IntsRef;
+import com.graphhopper.util.EdgeIteratorState;
+import com.graphhopper.util.PMap;
+import com.graphhopper.util.PointList;
+
+import java.util.*;
+
+import static com.graphhopper.routing.util.EncodingManager.getKey;
+import static com.graphhopper.routing.util.PriorityCode.*;
+
+/**
+ * A flag encoder for wheelchairs.
+ * <p>
+ *
+ * @author don-philipe
+ */
+public class WheelchairFlagEncoder extends FootFlagEncoder {
+    final Set<String> excludeSurfaces = new HashSet<>();
+    final Set<String> excludeSmoothness = new HashSet<>();
+    final int maxInclinePercent = 6;
+    final int smallInclinePercent = 3;
+    final int maxKerbHeightCm = 3;
+
+    /**
+     * Should be only instantiated via EncodingManager
+     */
+    public WheelchairFlagEncoder() {
+        this(4, 1);
+    }
+
+    public WheelchairFlagEncoder(PMap properties) {
+        this((int) properties.getLong("speedBits", 4),
+                properties.getDouble("speedFactor", 1));
```

Comment:
Sorry, there is a bug in Foot and HikeFlagEncoder: can you use speed_bits and speed_factor instead of camel case here?